docs/proposals/202012_receive_split.md

```diff
@@ -8,30 +8,74 @@ status: accepted
+
 
 ### Goals
 
-1. Split the receive component into receive-route and receive (and ensure ease of resharding events).
-2. Evaluate any effects on performance by simulating scenarios and collecting and analyzing metrics.
-3. Use consistent hashing to avoid reshuffling time series after resharding events. The exact consistent hashing mechanism to be used needs some further research.
+* Reduce downtime of the ingestion logic in Thanos Receiver
+
+### Proposal
+
+We propose allowing to run Thanos Receiver in a mode that only forwards/replicates remote write (distributor mode). You can enable that mode by simply not specifying:
+
+```yaml
+  --receive.local-endpoint=RECEIVE.LOCAL-ENDPOINT
+                                 Endpoint of local receive node. Used to
+                                 identify the local node in the hashring
+                                 configuration.
+```
+
+We can call this mode a "Routing Receiver". Similarly, we can skip specify any hashring to Thanos Receiver (`--receive.hashrings-file=<path>`), explicitly purposing it only for ingesting. We can call this mode "Ingesting Receiver".
+
+User can also mix all of those two modes for various federated hashrings etc. So instead of what we had before:
+
+![Before](https://docs.google.com/drawings/d/e/2PACX-1vTfko27YB_3ab7ZL8ODNG5uCcrpqKxhmqaz3lW-yhGN3_oNxkTrqXmwwlcZjaWf3cGgAJIM4CMwwkEV/pub?w=960&h=720)
+
+We have:
+
+![After](https://docs.google.com/drawings/d/e/2PACX-1vTVrtCGjR4iMbrU7Kj6QAn1a1m4fr-kvoQVDAK4lzQ_wWfXfpLLEE9HB948-WHI5ZG6s1iGWt51R593/pub?w=960&h=720)
+
+This allows us to (optionally) model deployment in a way that avoid expensive re-configuration of the stateful ingesting Receivers after the hashring configuration file has changed.
+
+In comparison to previous proposal (as mentioned in [alternatives](#previous-proposal-separate-receive-route-command) we have big adventages:
+
+1. We can reduce number of components in Thanos system, we can reuse similar component flags and documentation. Users has to learn about one less command and in result Thanos design is much more approachable. Less components mean less maintainance, code and other implicit duties: Separate changelogs, issue confusions, boilerplates, etc.
+2. Allow consistent pattern with Query. We don't have separate StoreAPI component for proxying, we have that baked into Querier. This has been proven to be flexible and understandable, so I would like to propose similar pattern in Receiver.
+3. This is more future proof for potential advanced cases like chain of routers -> receivers -> routers -> receivers for federated writes, so trees with depth n.
+
+### Plan
+
+* Receiver without `--receive.hashrings` does not forward or replicate requests, it routes straight to multi-tsdb.
```

Comment:
Not sure about this suggestion, just that the old statement sounded contrapositive 😆  
```suggestion
* Receiver with `--receive.hashrings` forwards or replicate requests, without this, it would route the request straight to multi-tsdb.
```

Reply:
Resolved during office hours :)